tensorrt_llm/serve/scheduler.py

```diff
@@ -0,0 +1,217 @@
+import heapq
+from abc import ABC, abstractmethod
+from threading import Lock
+from typing import List
+
+from tensorrt_llm.serve.openai_protocol import CompletionRequest
+
+
+class Scheduler(ABC):
+
+    def __init__(self,
+                 ctx_servers: List[str] = None,
+                 gen_servers: List[str] = None):
+        self._ctx_servers = ctx_servers
+        self._gen_servers = gen_servers
+        self._lock = Lock()
+
+    def add_server(self, server: str, server_type: str):
+        raise NotImplementedError(
+            "Currently not support dynamic adding of servers")
+
+    def remove_server(self, server: str, server_type: str):
+        raise NotImplementedError(
+            "Currently not support dynamic removing of servers")
+
+    @abstractmethod
+    def schedule(self, request: CompletionRequest, server_type: str):
+        pass
+
+
+class RoundRobinScheduler(Scheduler):
+
+    def __init__(self,
+                 ctx_servers: List[str] = None,
+                 gen_servers: List[str] = None):
```

Comment:
I think we should have a router for ctx servers and a separate router for gen servers. This way you don't have to store both lists. 
I think we should have one router for ctx servers and one router for gen servers to simplify the code and make it more flexible (could you different routing strategies for ctx/gen servers.

```suggestion
    def __init__(self,
                 servers: List[str] = None,
```